pkg/route/apis/route/validation/validation.go

```diff
@@ -93,15 +92,19 @@ func validateRoute(route *routeapi.Route, check_hostname bool) field.ErrorList {
 	return result
 }
 
-// validateHost checks that a route's host name meets DNS requirements.
+// validateHost checks that a route's host name and subdomain meets DNS requirements.
 func validateHost(route *routeapi.Route) field.ErrorList {
-	result := field.ErrorList{}
-	if len(route.Spec.Host) == 0 {
-		return result
+
+	var result field.ErrorList
+	if len(route.Spec.Host) > 0 {
+		lenient, _ := route.Annotations[routev1.AllowNonDNSCompliantHostAnnotation]
+		result = append(result, library.ValidateHost(route.Spec.Host, lenient, field.NewPath("spec.host"))...)
+	}
+	if len(route.Spec.Subdomain) > 0 {
+		// Subdomain is not lenient because it was never used outside of routes
+		// TODO: fix library-go ValidateHost to take a boolean
+		result = append(result, library.ValidateHost(route.Spec.Host, "", field.NewPath("spec.subdomain"))...)
```

Comment:
```suggestion
		result = append(result, library.ValidateHost(route.Spec.Subdomain, "", field.NewPath("spec.subdomain"))...)
```
But `library.ValidateHost(route.Spec.Subdomain, "", ...` isn't right because it will not allow single-label subdomains.